src/shared/install.c

```diff
@@ -1708,36 +1708,58 @@ static int install_info_discover_and_check(
         return install_info_may_process(ret ? *ret : NULL, paths, changes, n_changes);
 }
 
-int unit_file_verify_alias(const UnitFileInstallInfo *i, const char *dst) {
+int unit_file_verify_alias(const UnitFileInstallInfo *i, const char *dst, char **ret_dst) {
+        _cleanup_free_ char *dst_updated = NULL;
         int r;
 
-        if (unit_name_is_valid(i->name, UNIT_NAME_TEMPLATE) &&
-            !unit_name_is_valid(dst, UNIT_NAME_TEMPLATE) &&
-            !i->default_instance)
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                       "Aliases for templates without DefaultInstance must be templates.");
+        const char *path_alias = strrchr(dst, '/');
+        if (path_alias) {
+                /* This branch covers legacy Alias= function of creating .wants
+                 * and .requires symlinks. */
+                _cleanup_free_ char *dir = NULL;
 
-        if (unit_name_is_valid(i->name, UNIT_NAME_INSTANCE | UNIT_NAME_TEMPLATE) &&
-            !unit_name_is_valid(dst, UNIT_NAME_TEMPLATE | UNIT_NAME_INSTANCE))
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                       "Aliases for template instances must be a templates or template instances containing the instance name.");
+                dir = dirname_malloc(dst);
```

Comment:
isn't this a bit sloppy? i.e. gets confused if i set SYSTEMD_UNIT_PATH=/wuff/miau.wants?

Reply:
This doesn't look at the actual path at all. This is for the case where a unit file has `[Install] Alias=foo.service.wants/something.target`, and we'd be looking at `foo.service.wants/something.target` here, and then check if `foo.service` is a valid unit name.